mirdata/datasets/egfxset.py

```diff
@@ -0,0 +1,387 @@
+
+    @property
+    def stringfret_tuple(self):
+        return self._track_metadata.get("String-fret Tuple")
+
+    @property
+    def note(self):
+        return self._track_metadata.get("Note")
+
+    @property
+    def midinote(self):
+        return self._track_metadata.get("Midinote")
+
+    @property
+    def pickup_configuration(self):
+        return self._track_metadata.get("Pickup Configuration")
+
+    @property
+    def effect(self):
+        return self._track_metadata.get("Effect")
+
+    @property
+    def model(self):
+        return self._track_metadata.get("Model")
+
+    @property
+    def effect_type(self):
+        return self._track_metadata.get("Effect Type")
+
+    @property
+    def knob_names(self):
+        return self._track_metadata.get("Knob Names")
+
+    @property
+    def knob_type(self):
+        return self._track_metadata.get("Knob Type")
+
+    @property
+    def setting(self):
+        return self._track_metadata.get("Setting")
+
+    @property
+    def audio(self) -> Optional[Tuple[np.ndarray, float]]:
+        """Solo guitar audio (mono)
+
+        Returns:
+            * np.ndarray - audio signal
+            * float - sample rate
+
+        """
+        return load_audio(self.audio_path)
+
+    def to_jams(self):
+        """Get the track's data in jams format
+
+        Returns:
+            jams.JAMS: the track's data in jams format
+
+        """
+        return jams_utils.jams_converter(
+            audio_path=self.audio_path, metadata=self._track_metadata
+        )
+
+
+@io.coerce_to_bytes_io
+def load_audio(fhandle: BinaryIO) -> Tuple[np.ndarray, float]:
+    """Load EGFxSet guitar audio
+
+    Args:
+        fhandle (str or file-like): File-like object or path to audio file
+
+    Returns:
+        * np.ndarray - audio signal
+        * float - sample rate
+
+    """
+    return librosa.load(fhandle, sr=None, mono=True)
+
+
+@core.docstring_inherit(core.Dataset)
+class Dataset(core.Dataset):
+    """
+    The EGFxSet dataset
+    """
+
+    def __init__(self, data_home=None, version="default"):
+        super().__init__(
+            data_home,
+            version,
+            name="egfxset",
+            track_class=Track,
+            bibtex=BIBTEX,
+            indexes=INDEXES,
+            remotes=REMOTES,
+            license_info=LICENSE_INFO,
+        )
+
+    @core.cached_property
+    def _metadata(self):
+        metadata_path = os.path.join(self.data_home, "egfxset_metadata.csv")
+        metadata_index = {}
+        tracknames = self.track_ids
+        cuerdas = {"1": 64, "2": 59, "3": 55, "4": 50, "5": 45, "6": 40}
+
+        try:
+            with open(metadata_path, "r") as fhandle:
+                csv_reader = csv.reader(fhandle, delimiter=",")
+                next(csv_reader)
+                for row in csv_reader:
+                    key = os.path.splitext(os.path.split(row[0])[1])[0]
+                    for track in tracknames:
```

Comment:
Hi, I've done all the changes already except this one, and the CSV in the test file, since we need it for the test. The main reason the dataset iterate through all the tracks is that our CSV does not have a list with track names, and the metadata is written by matching the track id's first letters of the effect to the CSV info. I have been trying to call this metadata outside the dataset definition like an older version of tinysol did but I haven't been successful at it. The other option is to change our metadata file to one that has all the track names (like tinysol does) and therefore remove the iteration.

Reply:
Thanks for the explanation. Can we use a dictionary instead?

Reply:
In the last push, I resolved the nested looping by having a small list that serves as an index of the rows in the CSV before going to the entire tracks to write the metadata. Is way faster than the panda solution and has passed all the checks. Do you think this is the right solution?